packages/integrations/image/src/vite-plugin-astro-image.ts

```diff
@@ -65,22 +64,17 @@ export function createPlugin(config: AstroConfig, options: Required<IntegrationO
 
 				meta.src = `__ASTRO_IMAGE_ASSET__${handle}__`;
 			} else {
-				const relId = path.relative(fileURLToPath(config.srcDir), id);
-
-				meta.src = join('/@astroimage', relId);
-
-				// Windows compat
-				meta.src = slash(meta.src);
+				meta.src = '/@astroimage' + url.pathname;
 			}
 
 			return `export default ${JSON.stringify(meta)}`;
 		},
 		configureServer(server) {
 			server.middlewares.use(async (req, res, next) => {
 				if (req.url?.startsWith('/@astroimage/')) {
-					const [, id] = req.url.split('/@astroimage/');
+					// Reconstructing URL to get rid of query parameters in path
+					const url = new URL(req.url.slice('/@astroimage'.length), 'file:');
 
-					const url = new URL(id, config.srcDir);
 					const file = await fs.readFile(url);
```

Comment:
If you are reviewing a PR, this is the core and the only meaningful change—the rest is just tests